03-Docker-Part-2/README.md

```diff
@@ -73,30 +76,92 @@ http localhost/version
 
 ```
 
-### Access from your local browser
+## Access from your local browser
 
 `Codespaces` can `forward` local ports so you can access remotely
 
-- Click on the `PORTS 4` tab
-- Choose `Add Port`
-- Enter 80
+- Click on the `PORTS` tab
 - Hover over `Local Address` for `Port 80` and click `Open in Browser`
 - You will get a new browser tab with the `Swagger UI`
   - Note: popup blockers may have to be disabled
 
-### Remove the web container
+## Generate web traffic
+
+We use [Web Validate (WebV)](https://github.com/microsoft/webvalidate) to generate and validate API requests 7 x 24
 
 ```bash
 
-docker rm web
+# verify you're in this directory
+pwd
+
+# run WebV --help
+docker run -it --rm ghcr.io/retaildevcrews/webvalidate --help
+
+# run a "baseline" test
+# -v mounts the current directory "into" the container
+# -s is our webserver
+# -f is the file in this directory
+docker run -it --rm --net web -v $(pwd):/app/TestFiles ghcr.io/retaildevcrews/webvalidate -s http://web:8080 -f baseline.json
+
+# check the web logs
+docker logs web
+
+# run 1 req/sec test
+# -v mounts the current directory "into" the container
+# --run-loop runs in a loop
+# --sleep waits 1000ms between request
+# --log-format json uses json logs
+# --verbose displays all logs (not just failures)
+# --prometheus exposes the Prometheus counters on /metrics
+docker run -d --name webv --rm --net web -p 8080:8080 -v $(pwd):/app/TestFiles ghcr.io/retaildevcrews/webvalidate -s http://web:8080 -f baseline.json --run-loop --sleep 1000 --log-format json --verbose --prometheus
```

Comment:
I think the "--rm" option needs to be removed. Currently when the container is stopped below with `docker stop webv`, the container is removed. Then the next command, `docker start webv` fails.

```suggestion
docker run -d --name webv --net web -p 8080:8080 -v $(pwd):/app/TestFiles ghcr.io/retaildevcrews/webvalidate -s http://web:8080 -f baseline.json --run-loop --sleep 1000 --log-format json --verbose --prometheus
```